File: C++/calc.cpp
/*
 * Christopher Bennett
 * Basic Text calculator in C++
 * Addition/Subtraction/Division/Multiplication
 * 21/07/2024
 * */


#include <iostream>



int main()
{
	std::cout << "\n\n\tWe Ball\n\n\n";

	double x, y;
	char operation;

	std::cout << "Enter Your First Number\n";
	std::cin >> x;
	std::cout << "Enter Your Second number\n";
	std::cin >> y;

	std::cout << "You Entered:\n";
	std::cout << x << "\n";
	std::cout << y << "\n";

	std::cout << "Enter A for Addition\n" << "Enter S for Subtraction\n";
	std::cout << "Enter D for Division\n" << "Enter M for multiplication\n";
	std::cin >> operation;
	operation = toupper(operation);

	std::cout << "You Entered\t" << operation << "\n";
	

	if (operation == 'A'){
		std::cout << "The Answer is\t" << x + y;
	}
	else if (operation == 'S'){
		std::cout << "The Answer is\t" << x - y;
	}
	else if (operation == 'D'){
		std::cout << "The Answer is\t" << x / y;
	}
	else if (operation == 'M'){
		std::cout << "The Answer is\t" << x * y;
	}
	std::cout << "\n";


	return 0;
}

File: Java/calc.java
import java.util.Scanner;



public class calc
{
	static char ChooseOperator()
	{
		char character = ' '; boolean isCorrectChar = false;
		Scanner sc = new Scanner(System.in); 

		while (!isCorrectChar)
		{	
			System.out.println("Enter an operator");
			try{
				character = sc.next().charAt(0);
				character  = Character.toUpperCase(character);
				isCorrectChar = true;
				if (!(Character.isDigit(character))){
					if ((character=='A') || (character=='S') || (character=='D') || (character=='M')){
						System.out.println("Entered a correct operation");
						isCorrectChar = true;
					}else{
						System.out.println("Enter one of the given letters");
						isCorrectChar = false;
					}
				}else if (Character.isDigit(character)){
					System.out.println("You Entered another noumber");
					isCorrectChar = false;
				}
			}catch(Exception e){
				System.out.println("Input One of the provid cahracters!");
				System.exit(0);
			}
		}
		return character;
	}
		

			
	static double Addition(double num1, double num2)
	{
		System.out.println("\n\t***ADDITION***");
		return (num1 + num2);
	}
	
	static double Subtraction(double num1, double num2)
	{
		System.out.println("\n\t***Subtraction***");
		return (num1 - num2);
	}
	
	static double Division(double num1, double num2)
	{
		System.out.println("\n\t***Divison***");
		return (num1 / num2);
	}

	static double Multiplication(double num1, double num2)
	{
		System.out.println("\n\t***Multiplication***");
		return (num1 * num2);
	}

	
	public static void main(String[] args)
	{
		Scanner keyboardInput = new Scanner(System.in);
		boolean isCorrectChar = false;
		char operator = ' ';
		
		System.out.println("Enter First Number");	
		double num1 = keyboardInput.nextDouble();	
		System.out.println("Enter Second Number");
		double num2 = keyboardInput.nextDouble();

		System.out.println("Enter A for Addition");
		System.out.println("Enter S for Subtraction");
		System.out.println("Enter D for Division");
		System.out.println("Enter M for division");

		operator = ChooseOperator();


		if (operator == 'A'){
			System.out.println(Addition(num1, num2));
		}
		else if (operator == 'S'){
			System.out.println(Subtraction(num1, num2));
		}
		else if (operator == 'D'){
			System.out.println(Division(num1, num2));
		}		
		else if (operator == 'M'){
			System.out.println(Multiplication(num1, num2));
		}
			
	}
}
		
		
